c++/misc.hpp

```diff
@@ -177,6 +177,29 @@ auto vector_of_keys(const std::map<K,V> &container)
 inline double atof(const std::string &s) { return std::atof(s.c_str()); }
 inline int atoi(const std::string &s) { return std::atoi(s.c_str()); }
 
+// Read data from stream F.
+template <typename T1, typename T2>
+void readtable(std::istream &F, std::vector<std::pair<T1, T2>> &v, bool verbose = false) 
+{
+    while (F) 
+    {
+        skip_comments(F);
+        T1 x;
+        T2 y;
+        F >> x >> y;
+        if (F.fail()) break;
+        assert(isfinite(x) && isfinite(y));
+        v.push_back(std::make_pair(x, y));
+    }
+    if (verbose) std::cout << v.size() << " lines read." << std::endl;
+}
+
+template <typename T1, typename T2>
+void writetable(std::vector<std::pair<T1, T2>> &re, std::ostream &F, int output_precision = 16) {
```

Comment:
re in output_precision naj bosta const. (Isto velja za verbose v readtable, in tudi sicer na sploh naj bodo programi const-correct.)
F.16: For “in” parameters, pass cheaply-copied types by value and others by reference to const